test/shopify-cli/task/ensure_env_test.rb

```diff
@@ -14,18 +14,148 @@ def setup
+        assert_equal response.first, EnsureEnv.call(@context)
+      end
+
+      def test_no_prompt_if_one_app_and_org
+        response = [{
+          "id" => 421,
+          "businessName" => "one",
+          "stores" => [{
+            "shopDomain" => "store.myshopify.com",
+          }],
+          "apps" => [{
+            "title" => "app",
+            "apiKey" => 1234,
+            "apiSecretKeys" => [{
+              "secret" => 1233,
+            }],
+          }],
+        }]
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(response)
+        Resources::EnvFile.any_instance.stubs(:write)
+        assert_equal response.first, EnsureEnv.call(@context)
+      end
+
+      def test_uses_default_values_for_env_file
+        Resources::EnvFile.expects(:parse_external_env).with.raises(Errno::ENOENT)
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(env_file_values).returns(stub(:write))
+        assert_equal partners_api_response[1], EnsureEnv.call(@context)
+      end
+
+      def test_keep_existing_env_values
+        Resources::EnvFile.expects(:parse_external_env).with.returns({ host: "host" })
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(env_file_values.merge({ host: "host" })).returns(stub(:write))
+        assert_equal partners_api_response[1], EnsureEnv.call(@context)
+      end
+
+      def test_not_all_required_values_found
+        Resources::EnvFile.expects(:parse_external_env).with.returns({ api_key: "apikey", secret: "secret" })
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(env_file_values).returns(stub(:write))
+        assert_equal partners_api_response[1], EnsureEnv.call(@context, required: [:api_key, :secret, :shop])
+      end
+
+      def test_all_required_values_found
+        Resources::EnvFile.expects(:parse_external_env).with.returns({ api_key: "apikey", secret: "secret" })
+        Resources::EnvFile.expects(:new).never
+        assert_empty(EnsureEnv.call(@context))
+      end
+
+      def test_regenerate_with_existing_env_file
+        Resources::EnvFile
+          .expects(:parse_external_env)
+          .returns({ api_key: "apikey", secret: "secret", shop: "shop", host: "host" })
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(env_file_values.merge({ host: "host" })).returns(stub(:write))
+        assert_equal partners_api_response[1], EnsureEnv.call(@context, regenerate: true)
+      end
+
+      def test_regenerate_no_existing_env_file
```

Comment:
nitpick: to be more accurate, this test seems to have an empty env file, and not "no existing env file".